### ISSUE TITLE
[BUG] `removeTilesOlderThan` removes tiles newer than timestamp

### ISSUE THREAD

**Issue by phong010198** [JaffaKetchup/flutter_map_tile_caching#172]
### What is the bug?

I'm using removeTilesOlderThan function to remove all tiles that were downloaded before `15 days ago` timestamp.
```dart
final mgmt = tileCachingStore.manage;
await mgmt.ready;
await mgmt.removeTilesOlderThan(
	expiry: DateTime.now().subtract(const Duration(days: 15)),
);
```
**Actual:** Remove all tiles that were downloaded (modified) AFTER timestamp.
**Expect:** Remove all tiles that were downloaded (modified) BEFORE timestamp.

In the docs:

> // Empty all tiles last modified before the specified timestamp

Function's in-line documentation:

> /// Remove tiles that were last modified after expiry from the specified store

### How can we reproduce it?

1. Download a polygon. `tileCachingStore.stats.all` return `(hits: 0, length: 100, misses: 0, size: 1478.74609375)`
2. Use removeTilesOlderThan
```dart
final mgmt = tileCachingStore.manage;
await mgmt.ready;
await mgmt.removeTilesOlderThan(
	expiry: DateTime.now().subtract(const Duration(days: 15)),
);
```
3. `tileCachingStore.stats.all` return `(hits: 0, length: 0, misses: 0, size: 0)`

### Platforms

Both Android and iOS

### Severity

Obtrusive: Prevents normal functioning but causes no errors in the console

**Comment by JaffaKetchup** [comment]
Hey @phong010198,
Thanks for reporting! I'll look into this.

**Comment by JaffaKetchup** [comment]
Thanks for reporting! This should be fixed in v9.1.4. Snuck in in a comparison operator bug.
